Source/Model/ModelParser.cs

```diff
@@ -305,6 +296,15 @@ internal override void Run()
                 }
               }
 
+              if (tuple0 == "else")
+              {
+                if (fn != null && !(resultName is string && ((string)resultName) == "#unspecified") && fn.Else == null)
```

Comment:
If it is guaranteed that fn != null at this line, then fn != null should be removed from the conditional.

Reply:
Also, if it turns out that tuple0 == "else" holds, then shouldn't it be guaranteed that fn.Else == null?  I suppose that there would be exactly one (or perhaps at most one) "else" in the model for a function.

Reply:
You are right. fn != null should be removed.
There should only be one `else` case in the boogie model. A check can be added to throw an exception if there are multiple `else` statements in the model. I will add this, shortly.